android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/MetadataResolver.java

```diff
@@ -236,6 +249,17 @@ public void onErrorResponse(VolleyError volleyError) {
     );
   }
 
+  public static boolean checkIfMetadataContainsSortingScores(Collection<UrlMetadata> urlMetadataCollection) {
```

Comment:
Ok that sounds reasonable. Then a TODO should be added.

On Tue Jan 27 2015 at 6:09:07 PM Hoà V. DINH <notifications@github.com>
wrote:

> In
> android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/MetadataResolver.java
> <https://github.com/google/physical-web/pull/277#discussion_r23661122>:
>
> > @@ -236,6 +249,17 @@ public void onErrorResponse(VolleyError volleyError) {
> >      );
> >    }
> >
> > +  public static boolean checkIfMetadataContainsSortingScores(Collection<UrlMetadata> urlMetadataCollection) {
>
> I think that's a fair concern.
>
> Keep in mind that it's a transitional state of the server. Eventually, the
> server will always send scores. Therefore, I think it would be overkill to
> create a new webservice to know the server capabilities.
>
> What we can do is making the server transition smooth for the client.
>
> I think we can proceed as follow:
>
>    - We can check in that temporary code on the client
>    - Move the server implementation to proxy sorting
>    - Then, clean up that temporary code and remove the client-side
>    sorting. We won't have to check whether the score is available or not.
>
> What do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/physical-web/pull/277/files#r23661122>.
>